### DISCUSSION TITLE
bug/performance fix suggestions

### DISCUSSION THREAD

**Discussion by Dan4oby** [ACGaming/UniversalTweaks#327] [Bugfix Suggestions]
https://bugs.mojang.com/browse/MC-123584?focusedId=461302&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-461302
https://bugs.mojang.com/browse/MC-33134 (spongeforge has fix of it)
https://bugs.mojang.com/browse/MC-101232

https://bugs.mojang.com/browse/MC-2536
https://bugs.mojang.com/browse/MC-51418

implent entity/tileentity activation range like in spigot performance patches. Spongeforge also has this patch. Vanilla 1.18 has simillar optimization named as simulation distance
https://docs.spongepowered.org/7.4.0/en/server/management/performance-tweaks.html

**Comment by jchung01** [comment]
MC-33134: SpongeForge's way of handling doesn't look like much of a "fix" as it (opt-in) disables saving of mineshafts to world data, but the [relevant code](https://github.com/SpongePowered/Sponge/blob/stable-7/src/main/java/org/spongepowered/common/mixin/optimization/world/gen/structure/MapGenStructureMixin_Structure_Saving.java#L106) and [context in the generated config under `structure-saving`](https://docs.spongepowered.org/stable/en/server/getting-started/configuration/sponge-conf.html)
MC-101232: Looks like it was [fixed by Forge](https://github.com/neoforged/NeoForge/commit/f81c0cb58fea2bd495528175035bb1dab3d93ffb) a long time ago.
